modules/free/arrow-free-data/src/test/kotlin/arrow/free/FreeApplicativeTest.kt

```diff
@@ -70,3 +71,8 @@ class FreeApplicativeTest : UnitSpec() {
     }
   }
 }
+
+private fun Gen.Companion.opsAp(gen: Gen<Int>) =
+  gen.map {
+    OpsAp.value(it)
+  }
```

Comment:
I'd love to make this one a recursive generator that can generate the whole opsAst, but kotlintest is no good at recursive gens, so I guess this is fine...

Reply:
would something like this be better?

```

fun Gen.Companion.opsAp() =
  Gen.oneOf(valueGen, addGen, subtractGen)

val valueGen = Gen.bind(Gen.int()) {
  OpsAp.value(it)
}

val addGen = Gen.bind(Gen.int(), Gen.int()) { a, b ->
  OpsAp.add(a, b)
}

val subtractGen = Gen.bind(Gen.int(), Gen.int()) { a, b ->
  OpsAp.subtract(a, b)
}
```

Reply:
Oh right, I somehow thought ops was a recursive type where `class Add(val a: Ops, val b: Ops)` or similar. Yes, that is much better

Reply:
ok, i have changed it